r/R/array.R

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#' Convert an object to a nanoarrow array
+#'
+#' @param x An object to convert to a array
+#' @param schema An optional schema used to enforce conversion to a particular
+#'   type. Defaults to [infer_nanoarrow_schema()].
+#' @param to A target prototype object describing the type to which `array`
+#'   should be converted.
+#' @param array An object of class 'nanoarrow_array'
+#' @param ... Passed to S3 methods
+#'
+#' @return An object of class 'nanoarrow_array'
+#' @export
+as_nanoarrow_array <- function(x, ..., schema = NULL) {
+  UseMethod("as_nanoarrow_array")
+}
+
+#' @rdname as_nanoarrow_array
+#' @export
+from_nanoarrow_array <- function(array, to = NULL, ...) {
+  stopifnot(inherits(array, "nanoarrow_array"))
+  UseMethod("from_nanoarrow_array", to)
+}
+
+#' @export
+as_nanoarrow_array.default <- function(x, ..., schema = NULL) {
+  # For now, use arrow's conversion for everything
```

Comment:
Most of the `as` functions call `arrow::something()`, is that temporary? What does this package do currently that doesn't require `arrow`?